### PR TITLE
[4.0] "Authentification" correction

### PR DESCRIPTION
**Pull request by particthistle**
Correction of language and hint positioning for two factor authentication field on login page.

Pull Request does not have a related issue - I identified issue while checking through translations, then found an issue where it appears in the site.

### Summary of Changes
- Correct "Authentification" to "Authentication" in COM_LOGIN_TWOFACTOR string (Authentification is valid, but not proper en_GB)
- On viewing where the string was being used, I then found the layout is incorrect, so have fixed that also.
- scss update to create new class for .form-control-hint to fix the layout.

### Testing Instructions
- In Joomla 4, enable one of the two factor authentication plugins
- Log out
- On the log in screen you'll now see the Secret Key box, with the spelling issue and the incorrect formatting of the hint that uses COM_LOGIN_TWOFACTOR string.
- Apply PR
- Log out again and check that the Log in screen now has the corrected en_GB spelling as well as better formatting per the AFTER result below.

### Actual result BEFORE applying this Pull Request
Before-autheitification
- Spelling error still visible - COM_LOGIN_TWOFACTOR string will display as "For Two-Factor Authentification" for en_GB
- Hint for Secret Key is supposed to be positioned on the right, but due to double span the text-right class fails to do so. 

### Expected result AFTER applying this Pull Request
image
- COM_LOGIN_TWOFACTOR string will now show as "For Two-Factor Authentication"
- Hint for Secret Key will now be displayed above the field, to the right of the field.

### Documentation Changes Required
No. en_GB translation will be updated, but will most likely have been correctly translated into other languages previously. I just spotted the issue when translating en_US.

### CODE REVIEW

**Comment by brianteeman** [comment]
LGTM

**Comment by particthistle** [comment]
I've checked I got the SCSS change correct this time by applying the PR package to my test site. Result was as expected in the AFTER result, so I look forward to seeing successful tests shortly.

**Comment by richard67** [comment]
@particthistle SCSS code style check fails in Drone, error log see here: https://ci.joomla.org/joomla/joomla-cms/36866/1/22

**Comment by richard67** [comment]
@particthistle Now drone says in the npm clone step:

```
CSS file copied/minified: /********/src/build/media_source/system/css/fields/calendar.css
--
Error: Invalid CSS after "}": expected "}", was ""
        on line 251 of administrator/templates/atum/scss/blocks/_login.scss
        from line 43 of administrator/templates/atum/scss/template.scss
>> }
```

The changes on GitHub for that file show up as:



Can it be that there is missing a closing curly bracket before the `.form-control-hint`?

Beside that, check code style. In scss we use 2 spaces for each level of indentation and no tabs.

If that helps, adjust your editor so it shows tabs and spaces, then you can see what is used.

File: administrator/templates/atum/scss/blocks/_login.scss [hunk 1]
```diff
@@ -60,6 +60,13 @@
           bottom: -1.1rem;
           font-size: .75rem;
           text-align: right;
+        }
+		.form-control-hint {
+          position: absolute;
+          top: .3rem;
+		  right: 0;  
+          font-size: .75rem;
+          text-align: right;
```
Comment (richard67) [inline administrator/templates/atum/scss/blocks/_login.scss]:
```suggestion
        }

        .form-control-hint {
          position: absolute;
          top: .3rem;
          right: 0;  
          font-size: .75rem;
          text-align: right;
```
Fix code style.

File: administrator/templates/atum/scss/blocks/_login.scss [hunk 2]
```diff
@@ -61,6 +61,13 @@
           font-size: .75rem;
           text-align: right;
         }
+        .form-control-hint {
+          position: absolute;
+          top: .3rem;
+          right: 0;
+          font-size: .75rem;
```
Comment (bembelimen) [inline administrator/templates/atum/scss/blocks/_login.scss]: Probably you should use this variable to be in sync: https://github.com/twbs/bootstrap/blob/main/scss/forms/_form-text.scss#L7

Comment (bembelimen) [reply administrator/templates/atum/scss/blocks/_login.scss]: See: https://github.com/joomla/joomla-cms/pull/31191

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 80e1daf4a70b7f2c869045e8413f0819e70421cf<br /><br />Tip: use a different browser to see Login form - no need to logout.
Needs npm ci
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31202">issues.joomla.org/tracker/joomla-cms/31202</a>.</sub>

File: administrator/templates/atum/scss/blocks/_login.scss (hunk 2 above)
Comment (particthistle) [reply administrator/templates/atum/scss/blocks/_login.scss]: #31191 was resolved by #31207. Changes made to _login.css in commit below follow variable structure from #31207 for all the font-size items in that file.

File: .gitignore [hunk 3]
```diff
@@ -107,3 +107,5 @@ RoboFile.ini
 # Media Manager
 /media/com_media/js/mediamanager.min.js.map
 /media/com_media/css/mediamanager.min.css.map
+administrator/templates/atum/scss/blocks/_notes/dwsync.xml
+administrator/templates/atum/scss/template.css
```
Comment (SharkyKZ) [inline .gitignore]: Looks like these were added by mistake.

```suggestion
```

Comment (particthistle) [reply .gitignore]: Can anyone point me to info on how to go about removing files from a commit in GitHub?

I had to redo the initial commits several times due to it wanting to merge the entire fork :(

**Comment by richard67** [comment]
@ceford Could you test again? Thanks in advance.

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 66dc2dcad7802b6968fce0319517f4085448d2a0<br /><br />Seems OK but I did not see the Web Authentication bar with either plugin.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31202">issues.joomla.org/tracker/joomla-cms/31202</a>.</sub>

**Comment by ceford** [comment]
screen shot 2020-11-01 at 19 03 07<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31202">issues.joomla.org/tracker/joomla-cms/31202</a>.</sub>

**Comment by richard67** [comment]
@ceford The web authentication button is only shown when accessing the site with https. When accessing with http it is not shown How did you test? https or http?.

**Comment by ceford** [comment]
Using localhost - and access is via http.

**Comment by richard67** [comment]
> Using localhost - and access is via http.

@ceford If localhost or other host doesn't matter. But as I wrote it is desired behaviour that the web authentication button is only shown with https and not with http.

**Comment by brianteeman** [comment]
More than desired it is intentional as webauthn will only work on https

**Comment by ceford** [comment]
Thanks for the tips! I did not have https enabled on localhost. I have now done that and see the Web Authentication button. Next job for me: find out what it does!

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 66dc2dcad7802b6968fce0319517f4085448d2a0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31202">issues.joomla.org/tracker/joomla-cms/31202</a>.</sub>

**Comment by zero-24** [comment]
> Next job for me: find out what it does!

Making your site more secure :-D You can check here as a starting point: https://en.m.wikipedia.org/wiki/WebAuthn

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31202">issues.joomla.org/tracker/joomla-cms/31202</a>.</sub>

**Comment by chmst** [comment]
Thanks!

**Pull request merged**